### ISSUE TITLE
Possible Security Issues.

### ISSUE THREAD

**Issue by rjbprime** [naturalcrit/homebrewery#926]
## Description

*TL;DR: This may not have anything to do with homebrewery itself, but it seems to be a security and/or a permissions issue, when the brew in question is apparently unpublished, but appearing in google search.*

I was searching on the PathfinderWiki Database, for the term ```whipspawn```, which I thought I originally got from there years ago. That term didn't come up, so I turned to trusty Google, and I was surprised to find, that excat term only appeared on google in brews **I'd** created on both homebrewery and GMBinder. So I logged into my homebrewery account, to check to see whether it was an unpublished brew, which it was, which, if they are unpublished, how in heck did google get a copy in it's archives? If they are unpublished, should they not be available for google to scrap?

I can prove that the guides in question (both the homebrewery version and the GMBinder version) were created by me. The account holder for the homebrewery version is RoximusJB, has a total of 27 brews, all unpublished. It does not seem that another of my campaign guides is available via google. Looking in an incognito window at my user profile, while not logged in, the brew section says, as it rightly should, ```No Brews.```

I'll be letting the GMBinder people know as well, as I swapped from from homebrewery about 2, 2 and a half years ago, when updates to homebrewery was up in the air, and if their system is showing the same issue, then I hope it can be rectified on both systems.

**Share Link** (not a homebrewery share link, but for reference) : https://www.google.com/search?q=whipspawn+tiefling&filter=0&biw=2744&bih=1398

**Comment by ericscheid** [comment]
This is a brew I've been working on, a simple investigation-style adventure: https://homebrewery.naturalcrit.com/share/rkWymyZP34

I've shared that link in various discords and fora for purpose of soliciting feedback. That is what the *Share* link is useful for. This brew is currently unpublished, but I am soliciting feedback from others via the share link.

What we have here then is a clash of definitions as to what "published" means. With many CMS a thing is "published" if it can be accessed anonymously (e.g. wordpress); while both homebrewery and gmbinder use the term "publish" as meaning "place the link where it can be seen" (i.e. in the published brews section of the user's home page). You can [see my published brews here](https://homebrewery.naturalcrit.com/user/erics), but you can also see any [unpublished brews for which I have shared a link](https://homebrewery.naturalcrit.com/share/rkWymyZP34) to.

Once I've shared the share link that link can travel, and others can discover that link. Including googlebot.

The issue has been noted and the first step is simply inserting a `<meta name="robots" content="noindex">` element for unpublished brews.

**Comment by rjbprime** [comment]
Fair enough. I was a bit spooked that something I thought I got from somewhere else and then trying to find the source idea of lead me back to my own creations, with said creations dominating the search term.

If its as simple as that, then it's all gravy. I'll just have to add that code to my brew from now on.

Thanks for the help, I'll close this.

**Comment by ericscheid** [comment]
That meta robots noindex code needs to be in the `<head>`, so adding into brews won't help =(

**Comment by rjbprime** [comment]
Then my next question is, how does one add it? Or is it needed to be enabled site wide?

**Comment by calculuschild** [comment]
This is something we will need to add to the site itself. I will see if I can get this working.

**Comment by calculuschild** [comment]
This should be handled now. It will take a while for google to update its cached pages but once it does your stuff shouldn't show up in Google search results anymore.
